src/components/map/toolbar/GeometryLog.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { observer } from 'mobx-react';
+import classnames from 'classnames';
+import { FaTimes } from 'react-icons/fa';
+import { FiClipboard } from 'react-icons/fi';
+import * as s from './geometryLog.scss';
+
+interface IGeometryLogState {
+    isToggled: boolean;
+}
+
+interface IGeometryLogProps {
+}
+
+@observer
+export default class GeometryLog extends React.Component<IGeometryLogProps, IGeometryLogState> {
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            isToggled: false,
+        };
+    }
+
+    private getLogs = () => {
+        return 'logs';
+    }
+
+    private toggleLog = () => {
+        const isToggled = !this.state.isToggled;
+        this.setState({ isToggled });
+    }
+
+    render() {
+        return (
```

Comment:
This might be simplified by having two separate returns

Reply:
Is it a readability issue? Maybe this would help:
```suggestion
(this.state.isToggled) ?
   this.logView() :
   <FiClipboard onClick={this.toggleLog}/>
```